console/console-init/ui/src/modules/device/dialogs/CreateDevice/CreateDevice.tsx

```diff
@@ -1,21 +1,174 @@
 
-export const CreateDevice: React.FunctionComponent<{}> = () => {
+const CreateDevice: React.FunctionComponent<any> = () => {
+  const [isWizardOpen, setIsWizardOpen] = useState<boolean>(true);
+  const [connectionType, setConnectionType] = useState<string>();
+  const onToggle = () => {
+    setIsWizardOpen(!isWizardOpen);
+    resetForm();
+  };
+  const resetForm = () => {};
+  const addGateway = {
+    name: "Add gateways",
+    component: <p>Add gateway</p>
+  };
+  const addCredentials = {
+    name: "Add credentials",
+    component: <p>Add credentials</p>
+  };
+  const reviewForm = {
+    name: "Review",
+    component: <p>Review</p>
+  };
+  const onConnectionChange = (_: boolean, event: any) => {
+    const data = event.target.value;
+    if (data) {
+      setConnectionType(data);
+    }
+  };
   const steps = [
-    { name: "Step 1", component: <DeviceInformation /> },
-    { name: "Step 2", component: <p>Step 2</p> },
-    { name: "Step 3", component: <p>Step 3</p> },
-    { name: "Step 4", component: <p>Step 4</p> },
+    { name: "Device information", component: <DeviceInformation /> },
     {
-      name: "Final Step",
-      component: <p>Final Step</p>,
-      hideCancelButton: true,
-      nextButtonText: "Close"
+      name: "Connection Type",
+      component: (
+        <ConnectionType
+          connectionType={connectionType}
+          onConnectionChange={onConnectionChange}
+        />
+      )
     }
   ];
+  if (connectionType) {
+    if (connectionType === "directly") {
+      steps.push(addCredentials);
+    } else {
+      steps.push(addGateway);
+    }
+    steps.push(reviewForm);
+  }
+
+  const handleNextIsEnabled = () => {
+    return false;
+  };
+
+  const CustomFooter = (
+    <WizardFooter>
+      <WizardContextConsumer>
+        {({ activeStep, onNext, onBack, onClose }) => {
+          if (
+            activeStep.name === "Device information" ||
+            activeStep.name === "Connection Type"
+          ) {
+            return (
+              <>
+                <Button
+                  variant="primary"
+                  type="submit"
+                  onClick={onNext}
+                  className={
+                    activeStep.name === "Connection Type" && !connectionType
+                      ? "pf-m-disabled"
+                      : ""
+                  }
+                >
+                  Next
+                </Button>
+                <Button
+                  variant="secondary"
+                  onClick={onBack}
+                  className={
+                    activeStep.name === "Device information"
+                      ? "pf-m-disabled"
+                      : ""
+                  }
+                >
+                  Back
+                </Button>
+                <Button variant="link" onClick={onClose}>
+                  Cancel
+                </Button>
+              </>
+            );
+          }
+          if (activeStep.name !== "Review") {
```

Comment:
can you try to use lowerCase.

Reply:
not possible as it directly reflects in the UI screen